src/pgen/cluster.cpp

```diff
@@ -405,4 +432,101 @@ void ProblemGenerator(MeshBlock *pmb, parthenon::ParameterInput *pin) {
+
+  // fill derived vars (*including ghost cells*)
+  auto &coords = pmb->coords;
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+
+  pmb->par_for(
+      "FillDerivedHydro", kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        // get gas properties
+        const Real rho = prim(IDN, k, j, i);
+        const Real v1 = prim(IV1, k, j, i);
+        const Real v2 = prim(IV1, k, j, i);
+        const Real v3 = prim(IV1, k, j, i);
+        const Real P = prim(IPR, k, j, i);
+
+        // compute radius
+        const Real x = coords.Xc<1>(i);
+        const Real y = coords.Xc<2>(j);
+        const Real z = coords.Xc<3>(k);
+        const Real r2 = SQR(x) + SQR(y) + SQR(z);
+        log10_radius(0, k, j, i) = 0.5*std::log10(r2);
+
+        // compute entropy
+        const Real K = P / std::pow(rho, gam);
+        entropy(0, k, j, i) = K;
+
+        const Real v_mag = std::sqrt(SQR(v1) + SQR(v2) + SQR(v3));
+        const Real c_s = std::sqrt(gam * P / rho); // ideal gas EOS
+        const Real M_s = v_mag / c_s;
+        mach_sonic(0, k, j, i) = M_s;
+
+        // compute temperature
+        temperature(0, k, j, i) = mbar_over_kb * P / rho;
+      });
+  if (pkg->Param<Cooling>("enable_cooling") == Cooling::tabular) {
+    auto &cooling_time = data->Get("cooling_time").data;
+
+    // get cooling function
+    const cooling::TabularCooling &tabular_cooling =
+        pkg->Param<cooling::TabularCooling>("tabular_cooling");
+    const auto cooling_table_obj = tabular_cooling.GetCoolingTableObj(pkg);
+
+    pmb->par_for(
+        "FillDerivedCoolingTime", kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+        KOKKOS_LAMBDA(const int k, const int j, const int i) {
+          // get gas properties
+          const Real rho = prim(IDN, k, j, i);
+          const Real P = prim(IPR, k, j, i);
+
+          // compute cooling time
+          const Real eint = P / (rho * gm1);
+          const Real edot = cooling_table_obj.DeDt(eint, rho);
+          const Real t_cool = eint / edot;
+          cooling_time(0, k, j, i) = t_cool;
+        });
+  }
+
+  if (pkg->Param<Fluid>("fluid") == Fluid::glmmhd) {
+    auto &plasma_beta = data->Get("plasma_beta").data;
+
+    pmb->par_for(
+        "FillDerivedMHD", kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+        KOKKOS_LAMBDA(const int k, const int j, const int i) {
+          // get gas properties
+          const Real rho = prim(IDN, k, j, i);
+          const Real Pgas = prim(IPR, k, j, i);
+          const Real B1 = prim(IB1, k, j, i);
+          const Real B2 = prim(IB2, k, j, i);
+          const Real B3 = prim(IB3, k, j, i);
+          const Real Pmag = 0.5 * (SQR(B1) + SQR(B2) + SQR(B3));
+
+          // compute plasma beta
+          plasma_beta(0, k, j, i) = Pgas / Pmag;
```

Comment:
Should check that if `Pmag == 0` we set it to sth tiny.

Reply:
As before, is NaN a bad answer? What would be more sensible?

Reply:
I think NAN would be good but hack in NAN exclusion into the Ascent 1d profiles and projects if that's possible

Reply:
Not sure if that's possible. We'd need to ask the Ascent devs.